rwengine/src/engine/GameData.cpp

```diff
@@ -409,6 +409,15 @@ ClumpPtr GameData::loadClump(const std::string& name) {
     return m;
 }
 
+ClumpPtr GameData::loadClump(const std::string& name, const std::string& textureSlot) {
+    std::string currentSlot = currenttextureslot;
+    if (!textureSlot.empty())
```

Comment:
Btw, does this `if` change anything? I mean you save current state of `currenttextureslot` in `currentSlot`. After that you override it or not. But at the end you use value from `currentSlot`. (or I don't notice something?)

Reply:
`currenttextureslot` is global variable. It used inside of loadClump to determine which texture slot to use to search for texture for this model




```c++
(GameData.cpp)
...
GameData::GameData(Logger* log, const rwfs::path& path)
    : datpath(path), logger(log), engine(nullptr) {
    dffLoader.setTextureLookupCallback(
        [&](const std::string& texture, const std::string&) {
            return findSlotTexture(currenttextureslot, texture);
        });
}
```

Reply:
Thx for explanation.